pkg/test/framework/suite.go

```diff
@@ -108,16 +109,32 @@ func (s *Suite) RequireEnvironment(name environment.Name) *Suite {
 		return nil
 	}
 
-	// Prepend the function, so that it runs as the first thing.
-	fns := []resource.SetupFn{setupFn}
-	fns = append(fns, s.setupFns...)
-	s.setupFns = fns
+	s.requireFns = append(s.requireFns, fn)
+	return s
+}
+
+// RequireClusters ensures that the current environment contains exactly the number of clusters expected.
+// Otherwise it stops test execution.
+func (s *Suite) RequireClusters(numClusters int) *Suite {
```

Comment:
Do we want to distinguish this explicitly? I was thinking its either single cluster or multi clusters, is there a case where we want a test to run with 3 clusters but not 4?

Reply:
probably not. I had considered making this a minimum. Or we could say that multicluster is always 3, but that seems like it might eventually be a limitation.

Thoughts?

FYI @ayj

Reply:
Minimum number should be fine. I don't think there is an issue with extra clusters if they go unused by the test.